tests/testthat/helper-TealAppDriver.R

```diff
@@ -1,5 +1,28 @@
-# Import non-exported TealAppDriver from `teal` package
-TealAppDriver <- getFromNamespace("TealAppDriver", "teal") # nolint: object_name.
+# Initialization function to create a new TealAppDriver object
+#
+# By manipulating the server function as below, we can hint shinytest2 to load
+# this package and its "Depends".
+# Related to https://github.com/rstudio/shinytest2/issues/381
+init_teal_app_driver <- function(...) {
+  shiny__shinyApp <- shiny::shinyApp # nolint: object_name.
+  testthat::with_mocked_bindings(
+    {
+      TealAppDriver <- getFromNamespace("TealAppDriver", "teal") # nolint: object_name.
+      TealAppDriver$new(...)
+    },
+    shinyApp = function(ui, server, ...) {
+      functionBody(server) <- bquote({
+        # Hint to shinytest2 that this package is should be available (via {globals})
+        .hint_to_load_package <- add_facet_labels
+        .(functionBody(server))
+      })
+
+      do.call(shiny__shinyApp, append(x = list(ui = ui, server = server), list(...)))
```

Comment:
what `shiny__shinyApp` sorcery is this : P

Reply:
hahahah :mage: 

We still want to call the same function that is being mocked, hence we keep it in memory and use it here.

I kept the same naming convention as in `{tmc}` where this method is also used

Reply:
Technically, we could just use `shiny::shinyApp` here as this changes the bindings in the `teal` namespace.

I made a change to avoid this sorcery. Thanks for pointing it out.

Reply:
awesome, thanks